sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -3843,8 +3843,8 @@ object TimeWindowing extends Rule[LogicalPlan] {
    * The windows are calculated as below:
    * maxNumOverlapping <- ceil(windowDuration / slideDuration)
    * for (i <- 0 until maxNumOverlapping)
-   *   windowId <- ceil((timestamp - startTime) / slideDuration)
-   *   windowStart <- windowId * slideDuration + (i - maxNumOverlapping) * slideDuration + startTime
+   *   lastStart <- timestamp - (timestamp - startTime + windowDuration) % windowDuration
```

Comment:
Hmm, I suppose this is the window start time of last window. Don't we need to consider slide duration when calculating last start? I think this is only correct if the slide duration equals to window duration.

Reply:
That's inconsistency between code and comment :) Nice finding.

Reply:
@nyingping Could you please fix the comment here? Thanks!

Reply:
@HeartSaVioR Oh, yea, I verified the calculation by looking at the comment. The code looks correct.

Reply:
The code comment is not fixed; please don't resolve the conversation manually. You can fix it, push the commit and the comment thread will become outdated.